addons/html_builder/static/src/plugins/background_option/background_shape_option_plugin.js

```diff
@@ -332,6 +348,41 @@ export class BackgroundShapeOptionPlugin extends Plugin {
         }
         return this.backgroundShapesById;
     }
+    handleBgColorChanged(editingElement, newColor = "") {
+        if (newColor) {
+            const style = getHtmlStyle(this.document);
+            const isColorCombination = /^o_cc[12345]$/.test(newColor);
+            if (isColorCombination) {
+                // Extract color from color combination class
+                // (e.g., 'o_cc1' -> '--o-cc1-bg')
+                const colorNumber = newColor.substring(4);
+                const colorVar = `o-cc${colorNumber}-bg`;
+                newColor = getCSSVariableValue(colorVar, style).toLowerCase();
+                newColor = rgbToHex(newColor);
+            } else if (isCSSVariable(newColor)) {
+                // Extract color from CSS variable (e.g., 'var(--o-color-1)')
+                const customProperty = newColor.match(/var\(--(.+?)\)/)[1];
+                newColor = getCSSVariableValue(customProperty, style).toLowerCase();
+                newColor = rgbToHex(newColor);
+            } else if (isCSSColor(newColor)) {
+                // Convert any CSS color (rgb, rgba, hex) to hex format
+                newColor = rgbToHex(newColor);
+            } else {
+                // If it's not a valid color format, skip processing
+                newColor = "";
+            }
+        }
+        const { shape } = this.getShapeData(editingElement);
+        if (shape.includes("html_builder/Connections/")) {
+            this.applyShape(editingElement, () => ({
+                colors: this.getImplicitColors(editingElement, shape),
+            }));
+        }
+        const elements = this.getNeighborShape(editingElement);
+        for (const element of elements) {
+            this.updateShapeForNeighbor(element, newColor);
+        }
+    }
```

Comment:
Add this in a method istead of calling `handleBgColorChanged` at `on_removed_handlers`, `on_moved_handlers`, etc...